File: c++/5 kyu RGB To Hex Conversion.cpp
using namespace std;
class RGBToHex
{
  public:
  static std::string rgb(int r, int g, int b){
      	stringstream ss;
  	char c[16]={'0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F'};
  	if(r<0)	r=0;
  	if(g<0)	g=0;
  	if(b<0)	b=0;
  	if(r>255)	r=255;
    if(g>255)	g=255;
  	if(b>255)	b=255;
   	ss<<c[(int)r/16]<<c[r%16]<<c[(int)g/16]<<c[g%16]<<c[(int)b/16]<<c[b%16];
	return ss.str();
  }
};

File: Kotlin/5kyu RGB To Hex Conversion.kt
fun rgb(r: Int, g: Int, b: Int): String {
  	var s = "";
  	var c=charArrayOf('0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F');
  	if(r<0)	s+="00";
  	else if(r>255) s+="FF"
	else s+=c[r/16].toString()+c[r%16]
	if(g<0)	s+="00";
  	else if(g>255) s+="FF"
	else s+=c[g/16].toString()+c[g%16]
	if(b<0)	s+="00";
  	else if(b>255) s+="FF"
	else s+=c[b/16].toString()+c[b%16]
	return s;
}
